app/controllers/api/v2/push_notifications_controller.rb

```diff
@@ -1,13 +1,19 @@
 class Api::V2::PushNotificationsController < Api::V2::BaseController
 
   def create
-    exists = NotificationToken.find_by(:token => params[:token], :user => current_user)
-    if exists.nil?
-      NotificationToken.create(:token => params[:token], :user => current_user)
+    token = NotificationToken.find_by(:token => params[:token])
+    auth_token = Tiddle::TokenIssuer.build.find_token(current_user, request.headers['X-User-Token'])
+    if token.nil?
+      NotificationToken.create(:token => params[:token], :user => current_user, :authentication_token => auth_token)
       render json: { message: "push notification token saved" }, status: 200 and return
     else
```

Comment:
Este else me parece que está de sobra (junto con ese nivel de anidación). Si `token.nil?` es _truthy_ entonces llegará al `render ... and return` y quedará hasta ahí la ejecución.

Reply:
Perfecto, corregido